tests/Persistence/SqlAllMappingsLookupTest.php

```diff
@@ -0,0 +1,230 @@
+			self::getTestSysop()->getUser()
+		);
+
+		if ( $mappingList != null ) {
+			$this->setMappings( $itemId, $mappingList );
+		}
+	}
+
+	private function createPropertyWithMappings( PropertyId $propertyId, ?MappingList $mappingList = null ): void {
+		WikibaseRepo::getEntityStore()->saveEntity(
+			new Property( $propertyId, null, 'testType' ),
+			'',
+			self::getTestSysop()->getUser()
+		);
+
+		if ( $mappingList != null ) {
+			$this->setMappings( $propertyId, $mappingList );
+		}
+	}
+
+	private function setMappings( EntityId $entityId, MappingList $mappingList ): void {
+		WikibaseRdfExtension::getInstance()->newMappingRepository( self::getTestSysop()->getUser() )->setMappings(
+			$entityId,
+			$mappingList
+		);
+	}
+
+	public function newSqlAllMappingsLookup(): SqlAllMappingsLookup {
+		return new SqlAllMappingsLookup(
+			$this->db,
+			WikibaseRdfExtension::SLOT_NAME,
+			WikibaseRepo::getEntityIdParser(),
+			WikibaseRdfExtension::getInstance()->newMappingListSerializer()
+		);
+	}
+
+	public function testUnmodifiedMappings(): void {
+		$allMappings = $this->newSqlAllMappingsLookup()->getAllMappings();
+
+		// TODO: database is polluted by other tests
+		// $this->assertCount( 4, $allMappings );
+
+		$this->assertNull(
+			$this->getMappingsForEntityId( new ItemId( 'Q99001' ), $allMappings )
+		);
+		$this->assertEquals(
+			[
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' )
+			],
+			$this->getMappingsForEntityId( new ItemId( 'Q99002' ), $allMappings )->asArray()
+		);
+		$this->assertEquals(
+			[
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
+				new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
+				new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+			],
+			$this->getMappingsForEntityId( new ItemId( 'Q99003' ), $allMappings )->asArray()
+		);
+
+		$this->assertNull(
+			$this->getMappingsForEntityId( new PropertyId( 'P99001' ), $allMappings )
+		);
+		$this->assertEquals(
+			[
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' )
+			],
+			$this->getMappingsForEntityId( new PropertyId( 'P99002' ), $allMappings )->asArray()
+		);
+		$this->assertEquals(
+			[
+				new Mapping( 'foo:foo', 'https://example.com/#foo1' ),
+				new Mapping( 'foo:bar', 'https://example.com/#bar1' ),
+				new Mapping( 'foo:baz', 'https://example.com/#baz1' )
+			],
+			$this->getMappingsForEntityId( new PropertyId( 'P99003' ), $allMappings )->asArray()
+		);
+	}
+
+	/**
+	 * @param EntityMappingList[] $allMappings
+	 */
+	private function getMappingsForEntityId( EntityId $entityId, array $allMappings ): ?MappingList {
+		foreach ( $allMappings as $entityMappings ) {
+			if ( $entityMappings->entityId->equals( $entityId ) ) {
+				return $entityMappings->mappingList;
+			}
+		}
+		return null;
+	}
+
+	public function testEntityModified(): void {
+		// TODO - return the same mappings
+	}
+
+	public function testEmptyMappings(): void {
+		// TODO - is slot with empty mappings == entity without the mappings slot ?
```

Comment:
I will follow up on this in a Discussion. It's not clear if there is a practical difference between Entites that never had mappings versus Entities that had mappings at some point but were removed. Technically the difference is the presence of an empty list Slot value versus no Slot value.